app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/core/DetailView.java

```diff
@@ -0,0 +1,89 @@
+package com.example.nvdovin.weatherapp.presentation.details.core;
+
+
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Color;
+import android.os.Bundle;
+import android.support.design.widget.CollapsingToolbarLayout;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout;
+
+import com.example.nvdovin.weatherapp.R;
+import com.example.nvdovin.weatherapp.presentation.details.DetailActivity;
+import com.example.nvdovin.weatherapp.presentation.details.adapter.MainRecyclerAdapter;
+import com.example.nvdovin.weatherapp.presentation.history.HistoryActivity;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class DetailView {
+
+    private static final String ARGS_KEY = "HISTORY_ARGS";
+    private static final String CITY_ID_KEY = "CITY_ID_KEY";
+
+    @BindView(R.id.detail_main_recycler_view)
+    RecyclerView mainRecycler;
+    @BindView(R.id.detail_history_button)
+    FloatingActionButton floatingActionButton;
+    @BindView(R.id.detail_main_collapsing_toolbar)
+    CollapsingToolbarLayout collapsingToolbarLayout;
+
+    private MainRecyclerAdapter mainRecyclerAdapter;
+    private Context context;
+    private DetailsPresenter detailsPresenter;
+    private View detailsView;
+
+    public DetailView(DetailActivity detailActivity,
+                      MainRecyclerAdapter mainRecyclerAdapter,
+                      DetailsPresenter detailsPresenter) {
```

Comment:
View must not know about the presenter. Only the Presenter knows about the view and the model.